### PR TITLE
build: handle split gopath

### PR DESCRIPTION
**Pull request by imsodin**
Follow up to #4425 to handle multiple gopaths.

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -218,10 +218,17 @@ func main() {
...
-		if filepath.Join(gopath, "src/github.com/syncthing/syncthing") != wd {
+		for _, p := range filepath.SplitList(gopath) {
+			p, _ = filepath.EvalSymlinks(p)
+			if filepath.Join(p, "src/github.com/syncthing/syncthing") == wd {
+				inside = true
+				break
+			}
+		}
+		if !inside {
 			fmt.Println("You are running this outside of GOPATH/src/github.com/syncthing/syncthing, this might cause failure!")
```
Comment (calmh) [inline build.go]: While you're at it, maybe tweak this to something softer and more grammatical? (Yes, me bikeshedding messages again)

`You seem to have GOPATH set but the Syncthing source not placed correctly within it. Proceed with caution.` or something?

Comment (imsodin) [reply build.go]: Sure thing. Please go ahead and bikeshed my language as much as you can. I am decent at correcting somebody else's messages but rubbish at producing good ones myself.

**Comment by calmh** [comment]
I don't have things set up to trivially pull in pull requests to my local source to I'm taking this on faith

**Comment by calmh** [comment]
@st-review merge

build: Handle split GOPATH

**Comment by st-review** [comment]
@calmh: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by calmh** [comment]
oh and please do correct capitalization of the commit subjects and we won't have to tweak them 💃

File: build.go [hunk 2]
```diff
@@ -218,11 +218,18 @@ func main() {
...
-			fmt.Println("You are running this outside of GOPATH/src/github.com/syncthing/syncthing, this might cause failure!")
+		for _, p := range filepath.SplitList(gopath) {
+			p, _ = filepath.EvalSymlinks(p)
+			if filepath.Join(p, "src/github.com/syncthing/syncthing") == wd {
+				inside = true
+				break
+			}
+		}
+		if !inside {
+			fmt.Println("You seem to have GOPATH set but the Syncthing source not placed correctly within it, which may cause problems.")
```
Comment (imsodin) [inline build.go]: I didn't like your "proceed with caution" part that well, as the user can't really do anything anymore at that point, cautiously or not (except for hitting ctrl-c in a hurry). I hope I didn't just make it worse.

**Comment by imsodin** [comment]
> oh and please do correct capitalization of the commit subjects and we won't have to tweak them 💃

I was wondering why you always included my commit message in the st-review command - I don't seem to be at my most attentive at the moment.

File: build.go (hunk 2 above)
Comment (calmh) [reply build.go]: It looks perfectly fine to me :)

**Comment by st-review** [comment]
:ok_hand: Merged as 46becc5338c7939866df2253674d1726488ecc23. Thanks, @imsodin!

**Pull request closed without merging**